geoengine/layers.py

```diff
@@ -456,6 +452,54 @@ def search(self, search_string: str, *,
 
         return listings
 
+    def get_or_create_unique_collection(
```

Comment:
This method is quite complex. Could you add a test that verifies that it works?

Reply:
i really don't want to build the mock thingy for that...